src/reconcile/rec_col.c

```diff
@@ -426,66 +950,128 @@ __wt_rec_col_fix(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REF *pageref)
     /* Update the counters. */
     __wt_rec_incr(session, r, entry, __bitstr_size((size_t)entry * btree->bitcnt));
 
+    /* If there are entries we didn't write timestamps for, aggregate a stable timestamp. */
+    if (r->aux_entries < r->entries || r->entries == 0) {
+        WT_TIME_WINDOW_INIT(&unpack.tw);
+        WT_TIME_AGGREGATE_UPDATE(session, &r->cur_ptr->ta, &unpack.tw);
+    }
+
     /* Write the remnant page. */
-    WT_RET(__wt_rec_split_finish(session, r));
+    WT_ERR(__wt_rec_split_finish(session, r));
 
-    return (0);
+err:
+    if (hs_cursor != NULL)
+        WT_TRET(hs_cursor->close(hs_cursor));
+
+    return (ret);
 }
 
 /*
- * __wt_rec_col_fix_slvg --
- *     Reconcile a fixed-width, column-store leaf page created during salvage.
+ * __wt_rec_col_fix_write_auxheader --
+ *     Write the auxiliary header into the page image.
  */
-int
-__wt_rec_col_fix_slvg(
-  WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REF *pageref, WT_SALVAGE_COOKIE *salvage)
+void
+__wt_rec_col_fix_write_auxheader(WT_SESSION_IMPL *session, WT_RECONCILE *r, uint32_t entries,
+  uint32_t auxentries, uint8_t *image, size_t size)
 {
     WT_BTREE *btree;
-    WT_PAGE *page;
-    uint64_t page_start, page_take;
-    uint32_t entry, nrecs;
+    uint32_t auxdataoffset, auxheaderoffset, bitmapsize, offset;
+    uint8_t *endp, *p;
 
     btree = S2BT(session);
-    page = pageref->page;
+    (void)size; /* only used in DIAGNOSTIC */
```

Comment:
For bad historic reasons, we have a macro for this, `WT_UNUSED()`.

Since there's only one caller of `__wt_rec_col_fix_write_auxheader()`, we could lift the assert out to the caller? (No strong feelings either way.)

Reply:
Lifting it ends up putting the knowledge that there's no auxiliary header when auxentries == 0 in rec_write.c, which isn't really optimal from a layering standpoint. That knowledge is already implicitly exposed (where the image size is set) but that's itself not really optimal.

(Basically, to lift the assert out of the caller you also need the test for auxentries == 0, at which point you might as well lift the test out entirely and skip the call when auxentries is 0. Oh, and this is the assertion on line 1047; the one immediately adjacent could be lifted out but there's not much point.)

Reply:
Agreed.